packages/core/admin/admin/src/pages/SettingsPage/pages/ApplicationInfosPage/index.js

```diff
@@ -16,26 +18,48 @@ import { Link } from '@strapi/design-system/Link';
 import { Button } from '@strapi/design-system/Button';
 import ExternalLink from '@strapi/icons/ExternalLink';
 import Check from '@strapi/icons/Check';
+import { useConfigurations } from '../../../../hooks';
 import Form from './components/Form';
+import { fetchProjectSettings, postProjectSettings } from './utils/api';
+import getFormData from './utils/getFormData';
 
 const permissions = [{ action: 'admin::project-settings.update', subject: null }];
 
 const ApplicationInfosPage = () => {
   const inputsRef = useRef();
+  const toggleNotification = useNotification();
   const { formatMessage } = useIntl();
+  const queryClient = useQueryClient();
   useFocusWhenNavigate();
   const appInfos = useAppInfos();
   const { shouldUpdateStrapi, latestStrapiReleaseTag, strapiVersion } = appInfos;
-  const [modifiedData] = useState({ menuLogo: undefined });
+  const { updateProjectSettings } = useConfigurations();
+
+  const { data } = useQuery('project-settings', fetchProjectSettings);
 
   const currentPlan = appInfos.communityEdition
     ? 'app.components.UpgradePlanModal.text-ce'
     : 'app.components.UpgradePlanModal.text-ee';
 
-  const handleSubmit = () => {
-    // form values received here
-    // to do - handle this data
-    // const data = inputsRef.current.getValues();
+  const submitMutation = useMutation(body => postProjectSettings(body), {
+    onSuccess: async ({ menuLogo }) => {
+      await queryClient.invalidateQueries('project-settings', { refetchActive: true });
+      updateProjectSettings({ menuLogo: menuLogo?.url });
+    },
+  });
+
+  const handleSubmit = async () => {
```

Comment:
If this function is `async`, don't you need to `await` something? 🤔

Reply:
You are right, nothing `async` called in here, removed the `await` ;)